From 4af83f980712deb8ea0c9b43211f1df4cfec6027 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Mon, 11 Sep 2023 11:14:49 +0200
Subject: [PATCH 613/954] fix(stm32mp2-fdts): keep ommanager for hyperflash

When compiling device tree for hyperflash boot, the DT node
ommanager@40500000 should be kept, it should then be under flag.
The node spi@40430000 should also be removed in that case.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 17506b5761d6f83f57bdea28f3280a2413e3908d fix(stm32mp2-fdts): keep ommanager for hyperflash]
Change-Id: I08a116c36476f021fc0b0da8830b3674a3925a3f
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/383663
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp25-bl2.dtsi | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/fdts/stm32mp25-bl2.dtsi b/fdts/stm32mp25-bl2.dtsi
index 59c365afb1..c7ea0f8784 100644
--- a/fdts/stm32mp25-bl2.dtsi
+++ b/fdts/stm32mp25-bl2.dtsi
@@ -7,10 +7,13 @@
 
 / {
 	soc@0 {
-#if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR
+#if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR && !STM32MP_HYPERFLASH
 		/delete-node/ ommanager@40500000;
 #else
 		ommanager@40500000 {
+#if STM32MP_HYPERFLASH
+			/delete-node/ spi@40430000;
+#endif
 			/delete-node/ spi@40440000;
 		};
 #endif
-- 
2.53.0

